lib/ansible/module_utils/basic.py

```diff
@@ -1778,7 +1778,16 @@ def _check_argument_values(self, spec=None, param=None):
                 continue
             if isinstance(choices, SEQUENCETYPE) and not isinstance(choices, (binary_type, text_type)):
                 if k in param:
-                    if param[k] not in choices:
+                    # Allow one or more when type='list' param with choices
+                    if isinstance(param[k], list):
+                        for item in param[k]:
+                            if item not in choices:
+                                choices_str = ", ".join([to_native(c) for c in choices])
```

Comment:
not a required change, but it seems a 'difference' comparison between lists would give you any passed params that are not valid choices in 1 shot and a single error message.